src/lib.rs

```diff
@@ -654,6 +654,29 @@ impl<L, R> Either<L, R> {
             Either::Left(l) => panic!("{}: {:?}", msg, l),
         }
     }
+
+    /// Converts to a common type
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use either::*;
+    /// // Both u16 and u32 can be converted to u64.
+    /// let left: Either<u16, u32> = Left(3u16);
+    /// assert_eq!(left.into_common::<u64>(), 3u64);
+    /// let right: Either<u16, u32> = Right(7u32);
+    /// assert_eq!(right.into_common::<u64>(), 7u64);
+    /// ```
+    pub fn into_common<T>(self) -> T
```

Comment:
Common is a mildly awkward name because on the type level we want a "common type of L, R" but on the value level there is just one value - a particular Left(x) or Right(y) that we want to convert into a z. I still haven't come up with anything better. I think this is good enough.

Reply:
The only other name I thought of is "middle", as something between "left" and "right". I'm not sure that pun/analogy is any better than "common" though.

Reply:
Just `.into::<T>()` is a great name :) unfortunately we shouldn't use it, shouldn't shadow `Into::into`.

Reply:
What about `.union::<T>()`?
That is, `T` is the unified type of `L` and `R` where `L: Into<T>, R: Into<T>`.

I was also thinking `.merge::<T>()` to imply that the left and right branches are merged into a single branch, but "merge" implies that both exist simultaneously instead of *either* one branch or the other.

Reply:
We do have a couple methods that convert to a single type, [`either`](https://docs.rs/either/latest/either/enum.Either.html#method.either) and [`either_with`](https://docs.rs/either/latest/either/enum.Either.html#method.either_with), so maybe `either_into` fits. For example, `Either<i32, f32>` could call `x.either_into::<f64>()`.

There's also [`into_inner`](https://docs.rs/either/latest/either/enum.Either.html#method.into_inner) for `Either<T, T>` where it's already one "inner" type. That lends a little weight to the `into_common` naming approach.

Reply:
Oh, I like `either_into`, actually. As a user, I think the name alone makes the behavior easy to understand (`into` is called on either of the values). `into_common` or `into_middle` implies it's being converted to a *type called `common` or `middle`* IMO (kind of like [`str`'s `into_string`](https://doc.rust-lang.org/std/primitive.str.html#method.into_string)). When `into` isn't suffixed with another word it implies that type inference will handle the type conversion.

That's how I personally read it at least, you'll probably want more feedback.

Reply:
FYI: In Scala, this function is called [merge](https://www.scala-lang.org/api/2.13.6/scala/util/Either$$MergeableEither.html) (Ignore the sub-typing shenanigans).

Reply:
Hmm, `merge_into<T>` sounds pretty good.

Reply:
I agree with that, `merge into the type T` is very fluent and idiomatic.  (I also liked `either_into` btw)

Reply:
@bluss, any further bikeshed preference?

Reply:
@cuviper I renamed it.